teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,45 +330,411 @@ The entrypointer image should also be able to see that this flag is set and acco
+
+Performance analysis:
+- Signing and verification operations are local operations and should be negligible (in microseconds).
+- Creating and deleting a SPIRE entry takes about a round trip time (RTT) to the SPIRE server (and RTT from SPIRE server to DataStore - disk or network). This is done once during the initialization of the TaskRun(in 100-500ms).
+- Obtaining an SVID, this takes a RTT to the SPIRE server (and RTT from SPIRE server to KeyManager - disk or network). 
+  - For the tekton-pipelines-controller, this is done whenever the certificate expires (usually hour(s) - configurable). Rather negligible.
+  - For each TaskRun pod, the entrypointer obtains the SVID. This should take a RTT with the SPIRE server. However, there is a caveat here, that the SPIRE server needs to recognize that an entry has been created in order to fulfill the SVID request. The consistency of this can take up to a maximum of 15 seconds (and this happens because entries are created just in time (JIT)), and is in the critical path of the entrypointer's signing mechanism.
+- Obtaining a Trust Bundle takes a RTT to the SPIRE server. The SPIRE server should have this ready to serve and only needs to refresh it once in a long while (~hours if not days) with the UpstreamAuthority.
+
+Overall, there is minimal performance impact, with a slight consideration for TaskRun pod cold start-up latency.
+
+## Design Details - Failure Conditions
+
+The main single point of failure (SPOF) around the signing and verification ecosystem is the SPIRE server. Prolonged downtime of a SPIRE server would lead to the inability to sign/verify.
+
+However, if the SPIRE server is only down momentarily, workloads who have obtained information from the SPIRE server would be able to continue operations for as long as their certificates are valid. In our particular implementation, this implies that:
+
+- The tekton-pipelines-controller and TektonChains would be able to continue verification of TaskRuns for certificate validity time
+- The TaskRun pods would be able to sign if they already obtained their SVIDs
+- No new TaskRun pods would be able to sign when the SPIRE server goes down - as new entries can't be created, and new SVIDs can't be minted
+
+Another failure could be the SPIRE agents (daemonset), which would result in not being able to obtains SVIDs for that particular node while it is down (with the similar caveats for temporary failure and cert lifetime).
+
+## Design Details - Verification policy
+
+When the verifiers are unable to verify a document, either because the hashes don't match up or it is unable to obtain the trust bundle, what should the desirable action be? There are several options here:
+
+- Stop execution of a TaskRun
+- Indicate that the TaskRun is no longer verifiable but continue execution
+
+While this is at an early stage, we opt for indicating that a document is not verifiable, future additions can include ability to configure the action to be taken.
```

Comment:
What happens if the result cannot be verified when the task producing that result is part of a pipeline? The pipeline execution still continues? A pipeline can have the dependent task which is consuming the task result or it could be possible that the task result is not consumed at all

Reply:
Yup! It will keep running and just add an indication of non-verifiable via the annotation!